src/plugins/gs-plugin-eos.c

```diff
@@ -760,12 +737,21 @@ gs_plugin_eos_blacklist_by_branch_if_needed (GsPlugin *plugin, GsApp *app)
 	const char *branch = NULL;
 	const char *origin = gs_app_get_origin (app);
 	GsPluginData *priv = NULL;
+	g_autoptr(GSettings) settings = NULL;
 
 	if (!gs_app_is_flatpak (app))
 		return FALSE;
 
 	priv = gs_plugin_get_data (plugin);
-	default_branch = g_hash_table_lookup (priv->default_branches, origin);
+
+	/* do we want to filter to the default branch */
+	if (!g_settings_get_boolean (priv->settings, "filter-default-branch"))
```

Comment:
That's a good point indeed. I added this because in principle this was meant to be a backport of Richard's patch upstream... then ended up being a complete different thing.

I'll remove that and also the new schema changes